tests/unit/resource/file_resource_tests.cpp

```diff
@@ -0,0 +1,83 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "resource_tests.hpp"
+
+#include "umpire/resource/FileMemoryResource.hpp"
+
+#include "umpire/util/Exception.hpp"
+
+#include <limits.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "gtest/gtest.h"
+
+TEST(ResourceTest, AllocateDeallocate)
+{
+  auto alloc = std::make_shared<umpire::resource::FileMemoryResource>(umpire::Platform::host, "TEST", 0, umpire::MemoryResourceTraits{});
```

Comment:
When you make the tests a `TYPED_TEST_P` they will be using the fixture in `resource_tests.hpp`. This means that `this->memory_resource` will be a pointer to the FileMemoryResource object. (so you don't need to create on in every test)